### PR TITLE
[Core Concept] CPI #166

### PR DESCRIPTION
**Pull request by ZYJLiu**
added CPI example with visuals using Anchor book as reference

### CODE REVIEW

File: docs/core-concepts/cpi.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      content: index,follow,noodp
+  - - meta
+    - name: googlebot
+      content: index,follow
+footer: MIT Licensed
+---
+
+# Cross Program Invocation (CPIs)
+
+Cross Program Invocations allow Solana programs to call instructions from other programs and enables Solana programs to interact with each other.
```
Comment (jacobcreech) [inline docs/core-concepts/cpi.md:42]:
```suggestion
Cross Program Invocation allow Solana programs to call instructions from other programs, enabling composability between Solana programs.
```

File: docs/core-concepts/cpi.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Facts
+
+::: tip Fact Sheet
+
+- Cross Program Invocations enable for programs to call instructions on other programs, allowing for the composibility of Solana programs.
```
Comment (jacobcreech) [inline docs/core-concepts/cpi.md:48]:
```suggestion
- Cross Program Invocation enable programs to call instructions on other programs, allowing for the composability across Solana programs.
```

File: docs/core-concepts/cpi.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+- Program Derived Addresess (PDAs) can be used by one program to sign for Cross Program Invocations of instructions on another program.
```
Comment (jacobcreech) [inline docs/core-concepts/cpi.md:49]:
```suggestion
- Program Derived Addresses (PDAs) can be used to invoke signatures across programs.
```

File: docs/core-concepts/cpi.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+![User Call Initialize Function](./cpi_2.png)
+
+A user calls the Initialize instruction on the Puppet Program.
+
+![Puppet Program creates Puppet Account](./cpi_3.png)
+
+The Puppet Program then creates a Puppet Account with an empty Data field.
+
+![User Calls Pull String Function](./cpi_4.png)
```
Comment (jacobcreech) [inline docs/core-concepts/cpi.md:83]: This diagram is confusing. Instructions, data, accounts, and programs are all represented the same. Maybe some differentiating factors between them to make it easier to visualize?

**Review by jacobcreech — CHANGES_REQUESTED** [review]
I think this is great content, but it may be missing a few things to make a good deep dive on CPI. Most notably, `Invoke vs invoke_signed` as requested in the original issue https://github.com/solana-developers/solana-cookbook/issues/166

Add that and the requested changes and I think we should be good.

File: docs/core-concepts/cpi.md (hunk 4 above)
Comment (ZYJLiu) [reply docs/core-concepts/cpi.md:83]: Updated image. Any better?

**Pull request closed without merging**